### PR TITLE
Use cabal-docspec for doctests

### PR DESCRIPTION
**Pull request by phadej**

### CODE REVIEW

File: trifecta.cabal [hunk 1]
```diff
@@ -112,18 +106,6 @@ library
       fail                      == 4.9.*,
       semigroups                >= 0.8.3.1 && < 1
 
-test-suite doctests
-  type:              exitcode-stdio-1.0
-  main-is:           doctests.hs
-  ghc-options:       -Wall -threaded
-  x-doctest-options: -fobject-code
```
Comment (RyanGlScott) [inline trifecta.cabal:119]: I recall that with `doctest`, we had to enable `-fobject-code` directly to support the use of unboxed tuples in `trifecta`. I know that `cabal-docspec` behaves similarly to as if you had `-fobject-code` enabled, but does it actually need to enable the flag here in order to function properly?

Comment (phadej) [reply trifecta.cabal:119]: Recall, `cabal-docspec` loads already compiled library, it doesn't interpret the library source. Thus it's always, kind of `-fobject-code`.

Comment (phadej) [reply trifecta.cabal:119]: TL;DR with `cabal-docspec` you don't need `--fast` nor `-fobject-code`.

**Comment by phadej** [comment]
For the record, if anyone wonders when this will be released: `trifecta` will need `lens-5` changes (`indexed-traversable`), so (hopefuly shortly) after `lens-5` is on Hackage.

**Pull request merged**